operator/pkg/client/spec_sasl.go

```diff
@@ -55,6 +55,33 @@ func (c *Factory) configureAdminSpecSASL(ctx context.Context, namespace string,
 	return "", "", "", fmt.Errorf("unsupported SASL mechanism: %s", spec.SASL.Mechanism)
 }
 
+func (c *Factory) configureSchemaRegistrySpecSASL(ctx context.Context, namespace string, spec *redpandav1alpha2.SchemaRegistrySpec) (username, password, token string, err error) {
```

Comment:
nit: Seems like it might be more ergonomic to return a `sr.ClientOpt` here? Is there a reason to not?